src/Microsoft.Diagnostics.NETCore.Client/DiagnosticsClient/DiagnosticsClient.cs

```diff
@@ -98,8 +98,9 @@ public void WriteDump(DumpType dumpType, string dumpPath, bool logDumpGeneration
             {
                 case DiagnosticsServerResponseId.Error:
                     uint hr = BitConverter.ToUInt32(response.Payload, 0);
-                    if (hr == (uint)DiagnosticsIpcError.UnknownCommand) {
-                        throw new PlatformNotSupportedException($"Unsupported operating system: {RuntimeInformation.OSDescription}");
+                    if (hr == (uint)DiagnosticsIpcError.UnknownCommand)
+                    {
+                        throw new UnsupportedCommandException($"Unsupported operating system: {RuntimeInformation.OSDescription}");
```

Comment:
This will need corresponding changes in the [unit tests]( https://github.com/dotnet/diagnostics/blob/master/src/tests/Microsoft.Diagnostics.NETCore.Client/WriteDumpTests.cs#L74) and in [dotnet-dump](https://github.com/dotnet/diagnostics/blob/master/src/Tools/dotnet-dump/Analyzer.cs#L105)

Reply:
@noahfalk I made fixes. However `diagnostics-public-ci (Build and Test Diagnostics CentOS_7 Build_Debug)` is running for a day. It isn't normal on my opinion.

Reply:
Yeah it appears there was some network connectivity issue on the worker machine. I just restarted the verification and hopefully the issue won't reoccur.
cc @hoyosjs

Reply:
Yeah, there's 3 issues in our CI resiliency and 2 in Azure Networking. I'll check tomorrow. Thanks for letting me know.